client/src/components/dataList/SpeciesInfo.js

```diff
@@ -0,0 +1,215 @@
+import { useEffect, useState } from "react";
+import React from "react";
+import { useReducer } from "react";
+
+// create useReducer for changing values
+const reducer = (state, action) => {
+    console.log(action, 'this is the action');
+    switch (action.type) {
+        case 'editID':
+            console.log('Logged if the editName action is being dispatched');
+            //it updates the name to the input value
+            return { ...state, id: action.payload };
+
+        case 'editCommon_name':
+            return { ...state, common_name: action.payload };
+
+        case 'editScientificName':
+            return { ...state, scientific_name: action.payload };
+
+        case 'editPopulation':
+            return { ...state, population: action.payload };
+
+        case 'editConservationStatus':
+            return { ...state, conservation_status: action.payload };
+        case 'editCreatedOn':
+            return { ...state, created_on: action.payload };
+        case 'clearForm':
+            return { id: "", common_name: '', scientific_name: '', population: '', conservation_status: '', created_on: '' };
+        default:
+            return state;
+    }
+};
+const SpeciesInfo = () => {
+    const [species, setSpecies] = useState([]);
+
+    //get species data table
+    const getSpecies = async () => {
+        const response = await fetch(
+            `http://localhost:4040/species`
+        );
+        const specie = await response.json();
+        console.log(specie);
+        setSpecies(specie);
+    };
+
+    useEffect(() => {
+        getSpecies();
+    }, []);
+    //initialistate of the form will be empty
+    const initialState = {
+        id: "",
+        common_name: '',
+        scientific_name: '',
+        population: '',
+        conservation_status: '',
+        created_on: ''
+    };
+
+    const [state, dispatch] = useReducer(reducer, initialState);
+    console.log(state);
+
+    const handleAddSpecie = async (e) => {
+        e.preventDefault();
+
+        const newSpecie = {
+            id: state.id,
+            common_name: state.common_name,
+            scientific_name: state.scientific_name,
+            population: state.population,
+            conservation_status: state.conservation_status,
+            created_on: state.created_on
+        }
+        console.log(newSpecie);
+        const response = await fetch('http://localhost:4040/species',{
+            method: 'POST',
+            headers: {
+                Accept: 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(newSpecie)
+        });
+        const content = await response.json();
+        setSpecies([...species, content])
+        dispatch({type:'clearForm'});
+    }
+    return (
+        <>
```

Comment:
this is much easier to read (and easier to get the meaning from at a glance) than the sightings data table. This makes it easier to review!